### PR TITLE
Search: Included username and e-mail address in name search

### PR DESCRIPTION
**Pull request by amir-qayyum-khan**
#### What are the relevant tickets?
fixes https://github.com/mitodl/micromasters/issues/2668

#### What's this PR do?
Added email and username as search able fields.

#### How should this be manually tested?
**Run:** `docker-compose run web ./manage.py recreate_index`
GO to `http://192.168.99.100:8079/learners/` and in search box add email or user name

@pdpinch 
#### Screenshots (if appropriate)
screen shot 2017-03-01 at 1 29 46 pm

screen shot 2017-03-01 at 1 28 41 pm

screen shot 2017-03-01 at 1 29 37 pm

### CODE REVIEW

**Comment by noisecapella** [comment]
Does prefix search work with usernames? When I was working on a similar PR I needed to make an analyzed version of a field to allow `phrase_prefix` to work properly with it

**Comment by amir-qayyum-khan** [comment]
@noisecapella it was working fine, i test it manually.

**Comment by noisecapella** [comment]
Can you update the highlight to also cover the username?

File: static/scss/search-page.scss [hunk 1]
```diff
@@ -372,8 +372,18 @@
...
+        display: flex;
+        flex-direction: column;
+
+        .display-name {
+          font-size: 15px;
+          font-weight: 500;
+        }
+
+        .user-name {
+          color: #a0a0a0;
```
Comment (noisecapella) [inline static/scss/search-page.scss]: Please use the CSS variables for colors, like `$font-gray-light` here

File: search/indexing_api.py [hunk 2]
```diff
@@ -250,6 +250,7 @@ def program_enrolled_user_mapping():
             'school_state_or_territory': NOT_ANALYZED_STRING_TYPE,
         }},
         'email_optin': BOOL_TYPE,
+        'email': FOLDED_SEARCHABLE_STRING_TYPE,
```
Comment (noisecapella) [inline search/indexing_api.py]: Why not do this at the root level instead where email is already defined?

File: search/views_test.py [hunk 3]
```diff
@@ -212,7 +212,9 @@ def test_query_filters(self):
                     "fields": [
                         "profile.first_name.folded",
                         "profile.last_name.folded",
-                        "profile.preferred_name.folded"
+                        "profile.preferred_name.folded",
+                        "profile.username",
```
Comment (noisecapella) [inline search/views_test.py]: I am getting weird behavior when doing a prefix search on a username which includes dots. Can we create a folded version and use that to index?

**Comment by amir-qayyum-khan** [comment]
Done @noisecapella

File: static/js/components/search/LearnerResult.js [hunk 4]
```diff
@@ -41,17 +41,20 @@ class LearnerResult extends SearchkitComponent {
...
+          col={4}
+          className="learner-name"
           onMouseLeave={() => setLearnerChipVisibility(null)}
           onMouseEnter={() => setLearnerChipVisibility(profile.username)}
         >
           <span className="display-name">
             { highlight(getUserDisplayName(profile), this.searchkit.state.q) }
           </span>
+          <span className="user-name">
+            { profile.username }
```
Comment (noisecapella) [inline static/js/components/search/LearnerResult.js]: Can we highlight this value like we do for the other fields? See `display-name` above for an example

Comment (amir-qayyum-khan) [inline static/js/components/search/LearnerResult.js]: Done @noisecapella

**Comment by noisecapella** [comment]
Can you update the last test in `LearnerResult_test.js` to test for highlighting the username?

**Comment by amir-qayyum-khan** [comment]
Done @noisecapella

**Pull request merged**